doc: add note to described the possibility to use MultiNamespacedCach… (#2281)
operator-framework/operator-sdk: doc/user-guide.md

mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
```

> It is also possible to use the [MultiNamespacedCacheBuilder](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) to watch a specific set of namespaces:

Reviewer: We're using [reference style links](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#links) throughout the rest of the user-guide to make the raw markdown easier to read.
So perhaps change this to a reference link.

Author: done

```Go

> var namespaces []string // List of Namepsaces

Author: ```suggestion
var namespaces []string // List of Namespaces
```

> var namespaces []string // List of Namepsaces

Reviewer: Still a typo here:
```suggestion
var namespaces []string // List of Namespaces
```

Author: Done

// Create a new Cmd to provide shared dependencies and start components
mgr, err := manager.New(cfg, manager.Options{
   NewCache: cache.MultiNamespacedCacheBuilder(namespaces),

>    MapperProvider:     restmapper.NewDynamicRESTMapper,

Reviewer 2: Now that https://github.com/kubernetes-sigs/controller-runtime/pull/554 is merged and we've updated to v0.4.0, we should remove our dynamic rest mapper implementation and any references to it.

That can be done in a follow up though.

mgr, err := manager.New(cfg, manager.Options{Namespace: ""})
```

> **NOTE:** See [here](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) that for some scenarios is possible to inform a list of namespaces as well. 

Reviewer 2: ```suggestion
It is also possible to use the [MultiNamespacedCacheBuilder](https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/cache#MultiNamespacedCacheBuilder) to watch a specific set of namespaces:
```

And then perhaps include an example of how someone could watch 2 namespaces?
